gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -195,24 +221,29 @@ private SeekableByteChannel newWriteChannel(
             throw new UnsupportedOperationException(option.toString());
         }
       } else if (option instanceof CloudStorageOption) {
-        // These will be interpreted later.
+        // XXX: We need to interpret these later
       } else {
         throw new UnsupportedOperationException(option.toString());
       }
     }
-    CloudStoragePath cloudPath = checkPath(path);
-    if (cloudPath.seemsLikeADirectoryAndUsePseudoDirectories()) {
-      throw new CloudStoragePseudoDirectoryException(cloudPath);
+
+    if (!metas.isEmpty()) {
+      infoBuilder.metadata(metas);
     }
-    if (wantCreateNew) {
-      // XXX: Java's documentation says this should be atomic.
-      if (gcsService.getMetadata(cloudPath.getGcsFilename()) != null) {
-        throw new FileAlreadyExistsException(cloudPath.toString());
-      }
+    if (!acls.isEmpty()) {
+      infoBuilder.acl(acls);
+    }
+
+    try {
+      return new CloudStorageWriteChannel(gcsStorage.writer(infoBuilder.build(),
+          writeOptions.toArray(new Storage.BlobWriteOption[0])));
+    } catch (StorageException oops) {
+      // RPC API can only throw StorageException, but CloudStorageFileSystemProvider
+      // can only throw IOException. So we tunnel the latter inside the former.
+      Throwables.propagateIfInstanceOf(oops.getCause(), IOException.class);
+      if (oops.code()==404) throw new NoSuchFileException(oops.reason());
```

Comment:
space between '==' and use [braces](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml#L53) for the if condition.